packages/component-library/src/Button/index.tsx

```diff
@@ -0,0 +1,97 @@
+import clsx from "clsx";
+import type { ComponentProps, ComponentType, SVGProps } from "react";
+import { Button as BaseButton } from "react-aria-components";
+
+type Icon = ComponentType<SVGProps<SVGSVGElement>>;
+
+export type Props = Omit<ComponentProps<typeof BaseButton>, "children"> & {
+  variant?: "primary" | "secondary" | "solid" | "outline" | "ghost";
+  size?: "xs" | "sm" | "md" | "lg";
+  rounded?: boolean;
+  children?: string;
+  beforeIcon?: Icon;
+  afterIcon?: Icon;
+};
+
+export const Button = ({
+  className,
+  variant,
+  size,
+  rounded,
+  ...props
+}: Props) => (
+  <BaseButton
+    {...(variant &&
+      variant !== "primary" &&
+      !props.isPending &&
+      !props.isDisabled && { "data-variant": variant })}
+    {...(size && size !== "md" && { "data-size": size })}
```

Comment:
I would prefer the default ones be in the props. (i.e `size='md` as default and add `data-[size=md]` to classNames)

Reply:
oo, yeah thats fair I think I like that more too.  Will change to that.

Reply:
Done @keyvankhademi  -- I also restructured things a bit to add a `ButtonLink` component (i.e. looks the same as a button but used to link to another page and renders an `<a>` instead of a `<button`>